packages/guides/src/Nodes/DocumentNode.php

```diff
@@ -201,6 +205,17 @@ public function hasDocumentPart(string $identifier): bool
         return isset($this->documentPartNodes[$identifier]);
     }
 
+    public function addTocNode(TocNode $tocNode): void
```

Comment:
Why do we need this in a separate array? Is the function getNodes not enough to collect them from the tree?

Reply:
The thing is that the tocnodes might be in a section which is in a section which is in a section, so you have to traverse the whole tree to search for them, for which we would need to introduce several new methods